### DISCUSSION TITLE
Is it possible to support the Amazon Luna controller?

### DISCUSSION THREAD

**Discussion by BostonTodd** [darthcloud/BlueRetro#836] [Q&A]
I was able to pair the Luna controller to my BlueRetro Wireless PS2 adapter, but the buttons don't map correctly. I used an app in Free McBoot menu to test the controller, and found that the joysticks and d pad work, but not the buttons (except for the Menu button that somehow maps to the X button). I would be happy to map the buttons myself, if there is an app to do it... Let me know if I am asking in the wrong place. Thank you!

**Comment by darthcloud** [comment]
I can try if you get me a BT trace on Windows:
https://github.com/darthcloud/BlueRetro/wiki/Bluetooth-HCI-trace-with-Win10

**Comment by BostonTodd** [comment]
I think I did this correctly. Let me know!
[Amazon Luna Trace_TJM_05DEC23.zip](https://github.com/darthcloud/BlueRetro/files/13574950/Amazon.Luna.Trace_TJM_05DEC23.zip)

**Comment by BostonTodd** [comment]
Order of button press:
Buttons: Started at top and pressed clockwise (button labels are inverse from Switch pro). Top button (12 o'clock) is Y, then B (3 o'clock), A (6 O'clock) and X (9 o'clock).  Next the left joypad...same order 12 o'clock to 9 o'clock. Then the left joystick (same order) and right joystick (same order. then the Top left trigger, top right trigger, bottom left trigger, then bottom right trigger. Next the middle buttons...O (middle left), Menu (middle right) Microphone (middle bottom) and lastly, the main Luna button (large button in center...powers on the controller). Hope I got everything! Let me know
